src/test/java/org/folio/rest/api/InstanceBulkTest.java

```diff
@@ -0,0 +1,458 @@
+    URL getInstanceUrl = instanceBulkUrl("/ids?type=id&format=base64");
+
+    client.get(getInstanceUrl, TENANT_ID, json(getCompleted));
+
+    Response response = getCompleted.get(5, SECONDS);
+    validateMoonsResponse(response, expectedMatches, moons);
+  }
+
+  @Test
+  public void canGetInstanceBulkOfRawIdStream(TestContext context)
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+
+    int totalMoons = 500;
+    int expectedMatches = totalMoons;
+    Map<String, JsonObject> moons = manyMoons(totalMoons);
+    createManyMoons(moons);
+
+    URL getInstanceUrl = instanceBulkUrl("/ids?type=id&format=raw");
+
+    Buffer buffer = bufferedCheckURLs(context, getInstanceUrl, 200);
+
+    JsonObject collection = new JsonObject(buffer);
+    validateMoonsResponse(collection, expectedMatches, moons);
+  }
+
+  @Test
+  public void canGetInstanceBulkOfBase64IdStream(TestContext context)
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+
+    int totalMoons = 500;
+    int expectedMatches = totalMoons;
+    Map<String, JsonObject> moons = manyMoons(totalMoons,
+      InstanceBulkIdsGetField.ID,
+      InstanceBulkIdsGetFormat.BASE64);
+    createManyMoons(moons);
+
+    URL getInstanceUrl = instanceBulkUrl("/ids?type=id&format=base64");
+
+    Buffer buffer = bufferedCheckURLs(context, getInstanceUrl, 200);
+
+    JsonObject collection = new JsonObject(buffer);
+    validateMoonsResponse(collection, expectedMatches, moons);
+  }
+
+  @Test
+  public void canGetInstanceBulkOfRawIdWithQueryExact()
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+
+    int totalMoons = 10;
+    int expectedMatches = 1;
+    Map<String, JsonObject> moons = manyMoons(totalMoons,
+      InstanceBulkIdsGetField.ID,
+      InstanceBulkIdsGetFormat.RAW);
+    createManyMoons(moons);
+
+    String query = "query=keyword%20all%20%22Moon%20%233%22";
+    URL getInstanceUrl = instanceBulkUrl("/ids?type=id&format=raw&" + query);
+
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
+    client.get(getInstanceUrl, TENANT_ID, json(getCompleted));
+
+    Response response = getCompleted.get(5, SECONDS);
+    validateMoonsResponse(response, expectedMatches, moons);
+  }
+
+  @Test
+  public void canGetInstanceBulkOfRawIdWithQueryWildcard()
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+
+    int totalMoons = 20;
+    int expectedMatches = 11;
+    Map<String, JsonObject> moons = manyMoons(totalMoons,
+      InstanceBulkIdsGetField.ID,
+      InstanceBulkIdsGetFormat.RAW);
+    createManyMoons(moons);
+
+    String query = "query=keyword%20all%20%22Moon%20%231%2A%22";
+    URL getInstanceUrl = instanceBulkUrl("/ids?type=id&format=raw&" + query);
```

Comment:
Can you use `urlEncode` for readability?
Example: https://github.com/folio-org/mod-inventory-storage/blob/v18.2.3/src/test/java/org/folio/rest/api/ContributorTypesTest.java#L32

    String query = urlEncode("keyword all \"Moon #1*\"");
    URL getInstanceUrl = instanceBulkUrl("/ids?type=id&format=raw&query=" + query);

Reply:
The same for all other test methods with url encoded query strings.